packages/flutter/test/material/app_bar_test.dart

```diff
@@ -3416,12 +3416,37 @@ void main() {
         ),
       );
 
-      Color? leadingIconButtonColor() => iconStyle(tester, Icons.menu)?.color;
-      Color? actionIconButtonColor() => iconStyle(tester, Icons.add)?.color;
+      Color? leadingIconButtonColor() => _iconStyle(tester, Icons.menu)?.color;
+      Color? actionIconButtonColor() => _iconStyle(tester, Icons.add)?.color;
 
       expect(leadingIconButtonColor(), Colors.purple);
       expect(actionIconButtonColor(), Colors.purple);
     });
+
+    // This is a regression test for https://github.com/flutter/flutter/issues/130485.
+    testWidgets('Material3 - AppBar.iconTheme correctly applies custom white color in dark mode', (WidgetTester tester) async {
+      final ThemeData themeData = ThemeData(useMaterial3: true, brightness: Brightness.dark);
```

Comment:
Would it be better if we just use the ThemeData config mentioned in the issue description? Just feel it would be more straightforward:)
```dart
theme: ThemeData(
  colorScheme: const ColorScheme.dark().copyWith(onSurfaceVariant: Colors.red),
  useMaterial3: true,
)
```